QRCoder/Builders/Payloads/PayloadBase.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace QRCoder.Builders.Payloads
+{
+    public abstract class PayloadBase : IPayload, IConfigurableEccLevel, IConfigurableEciMode, IConfigurableVersion
+    {
+        protected virtual QRCodeGenerator.ECCLevel EccLevel { get; set; } = QRCodeGenerator.ECCLevel.Default;
+        QRCodeGenerator.ECCLevel IConfigurableEccLevel.EccLevel { get => EccLevel; set => EccLevel = value; }
```

Comment:
These interfaces are always explicitly implemented so that they do not appear on the intellisense list of accessible properties for the class.  Rather the extension method will appear, which uses generics to return the same type.

If that's confusing, for example, this code would break the fluent syntax:

```cs
public PayloadBase WithEciMode(EciMode value)
{
    _eciMode = value;
}
```

Because it returns the base class type, rather than the type of the derived type.  Extension methods allow the intended behavior.